crates/ironrdp-connector/src/lib.rs

```diff
@@ -97,6 +100,38 @@ pub struct Config {
     pub no_server_pointer: bool,
     /// If true, the INFO_AUTOLOGON flag is set in the [`ironrdp_pdu::rdp::ClientInfoPdu`].
     pub autologon: bool,
+    // for kerberos authentication
+    pub sspi_config: Option<SspiConfig>,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct SspiConfig {
+    pub kdc_url: Option<url::Url>,
+    pub hostname: Option<String>,
+}
+
+impl SspiConfig {
+    pub fn new(kdc_proxy_url: Option<impl Into<String>>, hostname: Option<impl Into<String>>) -> ConnectorResult<Self> {
```

Comment:
Overall, it’s not worth to have stuff like `Option<impl Into<String>>`
```suggestion
    pub fn new(kdc_proxy_url: Option<String>, hostname: Option<String>) -> ConnectorResult<Self> {
```